### ISSUE TITLE
Calling `_dealloc_warn` on a detached `TextIOWrapper` segfaults

### ISSUE THREAD

**Issue by devdanzin** [pypy/pypy#5123]
It's possible to segfault PyPy by first calling `detach`, then  `_dealloc_warn` of a `TextIOWrapper`:

```python
Python 3.10.14 (6b367d9e6715, Nov 14 2024, 04:48:15)
[PyPy 7.3.18-alpha0 with GCC 10.2.1 20210130 (Red Hat 10.2.1-11)] on linux
>>>> import sys
>>>> sys.__stdin__.detach()
<_io.BufferedReader name='<stdin>'>
>>>> sys.__stdin__._dealloc_warn(1)
Segmentation fault (core dumped)
```

**Comment by mattip** [comment]
Fixed in 7bc330cce0f.
